src/components/PrimaryPanes/SourcesTree.js

```diff
@@ -110,32 +107,26 @@ class SourcesTree extends Component<Props, State> {
     if (
       projectRoot != nextProps.projectRoot ||
       debuggeeUrl != nextProps.debuggeeUrl ||
-      thread != nextProps.thread ||
       nextProps.sourceCount === 0
     ) {
       // early recreate tree because of changes
       // to project root, debugee url or lack of sources
       return this.setState(
         createTree({
-          sources: nextProps.sources[nextProps.thread],
+          sources: nextProps.sources,
           debuggeeUrl: nextProps.debuggeeUrl,
           projectRoot: nextProps.projectRoot
         })
       );
     }
 
-    if (
-      nextProps.shownSource &&
-      nextProps.shownSource.thread == thread &&
-      nextProps.shownSource != shownSource
-    ) {
+    if (nextProps.shownSource && nextProps.shownSource != shownSource) {
```

Comment:
these checks can be simplified if shownSource and selectedSource or null if they belong to a different thread